frontend/src/general/__tests__/SnackbarMessage.test.jsx

```diff
@@ -0,0 +1,152 @@
+  it,
+  jest
+} from "@jest/globals";
+import { Snackbar, SnackbarContent, Box } from "@material-ui/core";
+import { CheckCircle, Error } from "@material-ui/icons";
+import SnackbarMessage from "../SnackbarMessage";
+
+Enzyme.configure({ adapter: new Adapter() });
+
+describe("<SnackbarMessage /> Class Component", () => {
+  let wrapper;
+
+  beforeEach(() => {
+    wrapper = Enzyme.shallow(<SnackbarMessage />);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    wrapper.unmount();
+  });
+
+  describe("render() function", () => {
+    describe("returns a component that", () => {
+      it("Contains 1 <Snackbar/> component with the expected props and 1 <SnackbarContent/> component", () => {
+        const snackbarComponent = wrapper.find(Snackbar);
+        expect(snackbarComponent).toHaveLength(1);
+
+        const snackbarComponentProps = snackbarComponent.props();
+        expect(snackbarComponentProps.contentprops).toEqual({
+          "aria-describedby": "message-id"
+        });
+        expect(snackbarComponentProps.anchorOrigin).toEqual({
+          vertical: "top",
+          horizontal: "right"
+        });
+        expect(snackbarComponentProps.open).toBe(wrapper.state().open);
+        expect(snackbarComponentProps.autoHideDuration).toEqual(3000);
+        expect(snackbarComponentProps.onClose).toBe(
+          wrapper.instance().handleClose
+        );
+
+        expect(wrapper.find(SnackbarContent)).toHaveLength(1);
+      });
+
+      describe("The <SnackbarContent/> component", () => {
+        describe("has a style prop", () => {
+          const statusColors = {
+            green: "#4caf50",
+            red: "#f44336"
+          };
+          describe("where if status", () => {
+            it("is success, the backgroundColor should be green", () => {
+              wrapper.setState({ status: "success" });
+              expect(wrapper.find(SnackbarContent).props().style).toEqual({
+                backgroundColor: statusColors.green
+              });
+            });
+
+            it("is anything else, the backgroundColor should be red", () => {
+              wrapper.setState({ status: "error" });
+              expect(wrapper.find(SnackbarContent).props().style).toEqual({
+                backgroundColor: statusColors.red
+              });
+            });
+          });
+        });
+        describe("and has a  message prop", () => {
+          describe("where if status", () => {
+            it("is success, it should have a <CheckCircle/> icon component", () => {
+              wrapper.setState({ status: "success" });
+              expect(wrapper.find(SnackbarContent).props().message).toEqual(
+                <Box id="message-id" className="snackMessage">
+                  <CheckCircle className="iconPadding" />
+                  {wrapper.state().message}
+                </Box>
+              );
+            });
+
+            it("is anything else, it should have an <Error/> icon component", () => {
```

Comment:
This phrasing makes it a bit disconnected and confusing - the subject of the sentence is in the previous block making the test name harder to read. Instead write "If status is ...."

Also, saying "anything else" is hard to understand without looking at the previous block. Instead write "If status is not success"

(I already suggested my preferred organizing for tests, just explicitly mentioning this here as a guideline for future tests)